src/NuGet.Core/NuGet.Commands/Utility/ContentHashUtility.cs

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Packaging;
+using NuGet.Packaging.Core;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+
+namespace NuGet.Commands.Utility
+{
+    public class ContentHashUtility : IContentHashUtility
```

Comment:
I' not sure about the purpose of this whole file... As I mentioned in my earlier review, you should just be able to use `.nupkg.metadata` file to get the content hash for each package from global packages folder. Only exception will be when someone used `DirectDownload` to install the package but that also isn't applicable for your current scenario (within VS). 

So it should be pretty straight forward, am I missing anything here?

Reply:
If I clear my global packages cache after installing a package, when I install a second package, the first package will not be re-extracted into the global packages folder, so the .nupkg.metadata file will not exist.

Let's talk offline.